Misc/NEWS.d/next/Library/2024-10-29-12-59-45.gh-issue-126033.sM3uCn.rst

```diff
@@ -0,0 +1,4 @@
+Fix a crash in :meth:`Element.remove <xml.etree.ElementTree.Element.remove>`
+when either the element to remove or one of the children implements an
+:meth:`~object.__eq__` method mutating its container node.
```

Comment:
It does not require implementing the `__eq__` method mutating its container node. It happens when the element is concurrently mutated.

Implementing the `__eq__` method is just the simplest way to test this.